book/art/fullnode.bob

```diff
@@ -1,27 +1,30 @@
-             .---------------------------.
-             |  Fullnode                 |
-             |                           |
- .--------.  |  .------------------.     |
- |        |---->|                  |     |
- | Client |  |  | JSON RPC Service |     |
- |        |<----|                  |     |
- `----+---`  |  `------------------`     |
-      |      |     ^                     |    .------------------.
-      |      |     |  .----------------. |    | Validators       |
-      |      |     |  | Gossip Service +----->|                  |
-      |      |     |  `--------+-------` |    |  .------------.  |
-      |      |     |         ^ |         |    |  |            |  |
-      |      |     |         | v         |    |  | Upstream   |  |
-      |      |  .--+---.   .-+---.       |    |  | Validators |  |
-      |      |  | Bank |<--| TVU |<--------------+            |  |
-      |      |  `------`   `-----`       |    |  `------------`  |
-      |      |     ^                     |    |                  |
-      |      |     |                     |    |  .------------.  |
-      |      |  .--+--.   .-----------.  |    |  |            |  |
-      `-------->| TPU +-->| Broadcast +--------->| Downstream |  |
-             |  `-----`   | Service   |  |    |  | Validators |  |
-             |            `-----------`  |    |  |            |  |
-             |                           |    |  `------------`  |
-             `---------------------------`    |                  |
-                                              `------------------`
-
+             .---------------------------------.
+             |  Fullnode                       |
+             |                                 |
+ .--------.  |  .------------------.           |
+ |        |---->|                  |           |
+ | Client |  |  | JSON RPC Service |           |
+ |        |<----|                  |           |
+ `----+---`  |  `------------------`           |
+      |      |     ^    .----------------.     |
+      |      |     |    | Gossip Service |     |  .------------------.
+      |      |     |    | (syncs votes & |<------>| Validators       |
+      |      |     |    |  contact info) |     |  |                  |
+      |      |     |    `----------------`     |  |                  |
+      |      |     |                           |  |  .------------.  |
+      |      |  .--+---.  .------.  .--------. |  |  |            |  |
+      |      |  | Bank |<-|Replay|  |Network |<------+ Upstream   |  |
+      |      |  | Forks|  |Stage |  |Observer| |  |  | Validators |  |
+      |      |  `------`  `--+---`  `-----+--` |  |  |            |  |
+      |      |   ^           |  ^         |    |  |  `------------`  |
+      |      |   | .------   |  |         |    |  |                  |
+      |      |   | |Gossip|<-` .+-----.   |    |  |                  |
+      |      |   | |Votes |    |ledger|<--`    |  |                  |
```

Comment:
Instead of ledger, how about Blocktree? Nearly everything in this diagram corresponds to the name of a struct in the codebase.

Reply:
👍